src/ApproxConv.jl

```diff
@@ -103,6 +103,8 @@ function prepareCommonConvWrapper!(ccwl::CommonConvWrapper{T},
     ccwl.cpt[i].p = collect(1:size(ccwl.cpt[i].X,1)) # collect(1:length(ccwl.cpt[i].Y))
     ccwl.cpt[i].Y = zeros(ccwl.xDim)  # zeros(ccwl.partial ? length(ccwl.usrfnc!.partial) : ccwl.xDim )
     ccwl.cpt[i].res = zeros(ccwl.xDim) # used in ccw functor for AbstractRelativeFactorMinimize
+    # TODO JT - Confirm it should be updated here.
+    ccwl.cpt[i].factormetadata.fullvariables = copy(Xi)
```

Comment:
is a copy necessary?  Factor operations should not be allowed to change the variable metadata?

Reply:
Copy was to be on the save side. I want to be sure the container does not change. It looks like it doesn't, but there are some comments in the code about constructing `Xi`:
> # TODO -- this build up of Xi is excessive and could happen at addFactor time
Perhaps `fullvairables` can be populated at construction time?

Reply:
@dehann perhaps rather in `prepgenericconvolution`? Or will it not make a difference.